app/features-json/setup_json_controller.rb

```diff
@@ -8,5 +8,109 @@ class SetupJSONController < APIController
+        json_error!(
+          error_message: "Missing required parameters #{missing_parameters.join(', ')}",
+          error_key: "Onboarding.Parameter.Missing"
+        )
+      end
+
+      dot_keys_values = {}
+
+      # Step: Validate & store the bot's GH API token
+      validate_api_token_correct!(bot_token)
+      dot_keys_values[:ci_user_api_token] = bot_token
+
+      # Step: Validate & store the bot's password
+      # TODO: We don't validate anything yet
+      dot_keys_values[:ci_user_password] = bot_password
+
+      # Step: Validate initial onboarding token
+      validate_api_token_correct!(initial_onboarding_token)
+      dot_keys_values[:initial_onboarding_user_api_token] = initial_onboarding_token
+
+      # Step: Validate encryption key
+      validate_encryption_key!(encryption_key)
+      dot_keys_values[:encryption_key] = encryption_key
+
+      # Step: Validate repo URL
+      # TODO: What do we want to validate here?
+      dot_keys_values[:repo_url] = config_repo
+
+      # The `.write_keys_file!` resets all services we already have running
+      # this is relevant when writing tests for this class
+      Services.dot_keys_variable_service.write_keys_file!(
+        locals: dot_keys_values
+      )
+
+      # After validating all the inputs, as well as storing all the inputs
+      # using the `dot_keys_variable_service` we can finally launch all
+      # the other services we need
+      Services.configuration_repository_service.setup_private_configuration_repo
+      Services.onboarding_service.clone_remote_repository_locally
+      Launch.start_github_workers
+
+      return json({})
+    end
+
+    #####################################################
+    # @!group Input validation
+    #####################################################
+
+    def validate_api_token_correct!(api_token)
+      if api_token.length != 40
+        json_error!(
+          error_message: "The GitHub token format is valid, they should be 40 characters long",
+          error_key: "Onboarding.Token.Invalid"
+        )
+      end
+
+      scope_validation_error = FastlaneCI::GitHubService.token_scope_validation_error(api_token)
+
+      unless scope_validation_error.nil?
+        scopes, required = scope_validation_error
+        scopes_list_wording = scopes.count > 0 ? scopes.map { |scope| "\"#{scope}\"" }.join(",") : "empty"
+        scopes_wording = scopes.count > 1 ? "scopes" : "scope"
+        error_message = "Token should include \"#{required}\" scope, currently it's" \
+                        " in #{scopes_list_wording} #{scopes_wording}."
+
+        json_error!(
+          error_message: error_message,
+          error_key: "Onboarding.Token.MissingScope"
+        )
+      end
+    end
+
+    def validate_encryption_key!(encryption_key)
+      if encryption_key.to_s.length < 5
+        json_error!(
+          error_message: "Encryption key not long enough, it must be at least 5 characters",
```

Comment:
oh i didn't know this was a requirement, we should add it to the UI to make sure it's at least 5 chars